From 8f195ec07ce0f5feb581b588f922704505ea346f Mon Sep 17 00:00:00 2001
From: Lu Hui <luhui@sipeed.com>
Date: Tue, 7 Nov 2023 16:05:14 +0800
Subject: [PATCH 12/56] drivers: aic8800: disable debug log


diff --git a/drivers/net/wireless/aic8800/aic8800_fdrv/Makefile b/drivers/net/wireless/aic8800/aic8800_fdrv/Makefile
index c39461efb..1ae275d89 100644
--- a/drivers/net/wireless/aic8800/aic8800_fdrv/Makefile
+++ b/drivers/net/wireless/aic8800/aic8800_fdrv/Makefile
@@ -130,7 +130,7 @@ CONFIG_RWNX_MON_RXFILTER ?= n
 
 # extra DEBUG config
 CONFIG_RWNX_SW_PROFILING ?= n
-CONFIG_RWNX_DBG ?= y
+CONFIG_RWNX_DBG ?= n
 BR_NAME = br0
 
 obj-$(CONFIG_AIC8800_WLAN_SUPPORT) := $(MODULE_NAME).o
-- 
2.17.1

