From 3c90865b8f4aad52469cbdb44aabb6648261ff91 Mon Sep 17 00:00:00 2001
From: Blue Cuenca <blue.cuenca@gmail.com>
Date: Sat, 28 Mar 2026 22:36:43 +0800
Subject: [PATCH 51/56] cedrus: reduce watchdog polling interval from 50ms to
 5ms


diff --git a/drivers/staging/media/sunxi/cedrus/cedrus_dec.c b/drivers/staging/media/sunxi/cedrus/cedrus_dec.c
index 193730af6..7291d6a27 100644
--- a/drivers/staging/media/sunxi/cedrus/cedrus_dec.c
+++ b/drivers/staging/media/sunxi/cedrus/cedrus_dec.c
@@ -108,7 +108,7 @@ void cedrus_device_run(void *priv)
 	if (!error) {
 		/* Start the watchdog timer. */
 		schedule_delayed_work(&dev->watchdog_work,
-				      msecs_to_jiffies(50));
+				      msecs_to_jiffies(5));
 
 		ctx->current_codec->trigger(ctx);
 	} else {
-- 
2.17.1

