From 882f27c40e16342a9c3e03f755f3ddf77e9e8cc6 Mon Sep 17 00:00:00 2001
From: Blue Cuenca <blue.cuenca@gmail.com>
Date: Sat, 28 Mar 2026 22:53:10 +0800
Subject: [PATCH 53/56] cedrus: watchdog back to 5ms (best practical with
 HZ=100)


diff --git a/drivers/staging/media/sunxi/cedrus/cedrus_dec.c b/drivers/staging/media/sunxi/cedrus/cedrus_dec.c
index 70de9b4e9..7291d6a27 100644
--- a/drivers/staging/media/sunxi/cedrus/cedrus_dec.c
+++ b/drivers/staging/media/sunxi/cedrus/cedrus_dec.c
@@ -108,7 +108,7 @@ void cedrus_device_run(void *priv)
 	if (!error) {
 		/* Start the watchdog timer. */
 		schedule_delayed_work(&dev->watchdog_work,
-				      msecs_to_jiffies(0));
+				      msecs_to_jiffies(5));
 
 		ctx->current_codec->trigger(ctx);
 	} else {
-- 
2.17.1

